pkg/querier/distributor_queryable.go

```diff
@@ -190,6 +192,9 @@ func (q *distributorQuerier) LabelNames(matchers ...*labels.Matcher) ([]string,
 }
 
 func (q *distributorQuerier) Close() error {
```

Comment:
I haven't take an in-depth look at this PR but I think it's worth confirming we make sure to call `Close()` on queriers everywhere. I seem to remember this wasn't always the case but we've gotten away with it because the existing ones don't leak anything.

Reply:
Found two more readers with a noop `Close()` method in our codebase. However those doesn't seem to be leaking anything, so I guess it's ok as they are. ([1](https://github.com/grafana/mimir/blob/main/pkg/querier/blocks_store_queryable.go#L432), [2](https://github.com/grafana/mimir/blob/main/pkg/frontend/querymiddleware/sharded_queryable.go#L146)).